src/utils.jl

```diff
@@ -0,0 +1,248 @@
+###############################################################################
+###############################################################################
+##
+##  versioninfo
+##
+###############################################################################
+###############################################################################
+# When a specific branch is loaded via `]add Package#branch` julia will only
+# create a checkout and keep a bare git repo in a separate directory.
+# In a bare repo HEAD will not point to the correct commit so we use the git
+# tree-hash that Pkg.jl provides and manually map this to a corresponding
+# commit.
+function _lookup_commit_from_cache(url::AbstractString, tree::AbstractString)
```

Comment:
I think also the const globals `jll_deps` and `cornerstones` should be moved to `utils.jl`.

(And perhaps it would make sense to split this into three files, say `utils/versioninfo.jl`, `utils/docs.jl`, `utils/tests.jl` or so -- but I am fine either way, just an idea)

Reply:
Another possible addition to the hypothetical `utils` dir might be `utils/rand.jl` with the `get_seeded_rng` function and friends, also from `Oscar.jl`.